test/unittest/tests/block/test_block_api_misc.cpp

```diff
@@ -0,0 +1,170 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+/*
+ * [block_api_misc]: block_mgr.c
+ * This file unit tests the miscellaneous block manager API functions:
+ *     - bm->addr_invalid
+ *     - bm->addr_string
+ *     - bm->block_header
+ *     - bm->is_mapped
+ *     - bm->size
+ *     - bm->stat
+ */
+
+#include <catch2/catch.hpp>
+#include <string>
+
+#include "wt_internal.h"
+#include "../wrappers/mock_session.h"
+
+TEST_CASE("Block manager invalid address", "[block_api_misc]")
+{
+    std::shared_ptr<mock_session> session = mock_session::build_test_mock_session();
+    WT_BLOCK b;
+    b.allocsize = 2;
+    b.objectid = 5;
+
+    // Pack all the stuff into an address cookie.
+    uint8_t p[WT_BTREE_MAX_ADDR_COOKIE], *pp;
+    pp = p;
+    REQUIRE(__wt_block_addr_pack(&b, &pp, WT_TIERED_OBJECTID_NONE, 10, 4, 12345) == 0);
+    size_t addr_size = WT_PTRDIFF(pp, p);
+
+    WT_BM bm;
+    WT_BM *bmp;
+    bmp = &bm;
+    __wt_bm_method_set(bmp, false);
+    bmp->block = &b;
+    bmp->is_live = false;
+
+    WT_BLOCK_CKPT ci;
+    WT_CLEAR(ci);
+    bmp->block->live = ci;
+    REQUIRE(__wt_spin_init(session->get_wt_session_impl(), &b.live_lock, "block manager") == 0);
+    REQUIRE(__wti_block_ckpt_init(session->get_wt_session_impl(), &bmp->block->live, "live") == 0);
+    SECTION("Test valid address")
+    {
+        pp = p;
+        REQUIRE(bmp->addr_invalid(bmp, session->get_wt_session_impl(), pp, addr_size) == 0);
+    }
+
+    SECTION("Test addr string")
+    {
+        WT_ITEM buf;
+        WT_CLEAR(buf);
+        pp = p;
+        REQUIRE(bmp->addr_string(bmp, session->get_wt_session_impl(), &buf, pp, addr_size) == 0);
```

Comment:
Is it possible for this to error out? Could we test case that?

Reply:
I would recommend that this have it's own test case for this particular function.